src/Acuminator/Acuminator.Analyzers/StaticAnalysis/EventHandlerModifier/EventHandlerProtectedModifierFix.cs

```diff
@@ -0,0 +1,193 @@
+
+					var localizedText = SyntaxFactory.Token(accessibilityModifier).Text;
+					if (addVirtualModifier)
+					{
+						localizedText += " " + SyntaxFactory.Token(SyntaxKind.VirtualKeyword).Text;
+					}
+
+					var makeProtectedTitle = nameof(Resources.PX1077Fix).GetLocalized(localizedText).ToString();
+					var codeFixAction = new ChangeAccessibilityModifierAction(makeProtectedTitle, context.Document, node, accessibilityModifier, addVirtualModifier);
+
+					context.RegisterCodeFix(codeFixAction, diagnostic);
+				}
+			}
+		}
+
+		private static bool GetPropertyValue(Diagnostic diagnostic, string propertyName)
+		{
+			if (diagnostic.TryGetPropertyValue(propertyName, out string? strValue) && bool.TryParse(strValue, out bool value))
+			{
+				return value;
+			}
+
+			return false;
+		}
+
+		internal class ChangeAccessibilityModifierAction : CodeAction
+		{
+			private readonly string _title;
+			private readonly Document _document;
+			private readonly MethodDeclarationSyntax _method;
+			private readonly SyntaxKind _accessibilityModifier;
+			private readonly bool _addVirtual;
+
+			private static readonly List<SyntaxKind> SyntaxKinds =
+			[
+				SyntaxKind.PrivateKeyword,
+				SyntaxKind.PublicKeyword,
+				SyntaxKind.ProtectedKeyword,
+				SyntaxKind.InternalKeyword,
+				SyntaxKind.VirtualKeyword,
+				SyntaxKind.OverrideKeyword
+			];
+
+			public override string Title => _title;
+			public override string EquivalenceKey => _title;
+
+			public ChangeAccessibilityModifierAction(string title, Document document, MethodDeclarationSyntax method, SyntaxKind accessibilityModifier, bool addVirtual)
+			{
+				_title = title;
+				_document = document;
+				_method = method;
+				_accessibilityModifier = accessibilityModifier;
+				_addVirtual = addVirtual;
+			}
+
+			protected override async Task<Document> GetChangedDocumentAsync(CancellationToken cancellationToken)
+			{
+				var index = -1;
+				for (var i = 0; i < _method.Modifiers.Count; i++)
+				{
+					if (SyntaxKinds.Contains(_method.Modifiers[i].Kind()))
+					{
+						index = i;
+						break;
+					}
+				}
+
+				var newToken = SyntaxFactory.Token(_accessibilityModifier);
+				if (index > -1)
+				{
+					newToken = newToken.WithTriviaFrom(_method.Modifiers[index]);
+				}
+
+				var newModifiers = _method.Modifiers.Where(m => !SyntaxKinds.Contains(m.Kind()));
+
+				var syntaxModifiers = SyntaxFactory.TokenList(newToken);
+
+				if (_addVirtual)
+				{
+					syntaxModifiers = syntaxModifiers.Add(SyntaxFactory.Token(SyntaxKind.VirtualKeyword));
+				}
+
+				syntaxModifiers = syntaxModifiers.AddRange(newModifiers);
+
+				var localDeclaration = _method.WithModifiers(syntaxModifiers);
+
+				var oldRoot = await _document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+				var newRoot = oldRoot!.ReplaceNode(_method, localDeclaration);
+
+				return _document.WithSyntaxRoot(newRoot);
+			}
+		}
+
+		internal class RemoveExplicitInterfaceAction : CodeAction
```

Comment:
We don't need the code action for the explicit interface removal. I described the reasons in the comment https://github.com/Acumatica/Acuminator/pull/554/files#r1757543094

Reply:
Done.